thunx-spring-api/src/main/java/com/contentgrid/thunx/spring/data/rest/AbacRequestFilter.java

```diff
@@ -37,7 +39,11 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
             log.debug("ABAC Context: {}", abacExpression);
             AbacContext.setCurrentAbacContext(abacExpression);
         } else {
-            log.warn("No X-ABAC-Context context present.");
+            var message = "No X-ABAC-Context context present.";
+            log.warn(message);
+            if (!allowMissingAbac) {
+                throw new IllegalStateException(message);
```

Comment:
> *IllegalStateException*
> Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation.
> -- https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalStateException.html

`IllegalStateException` is not an appropriate exception here. There is nothing wrong with the state of this object to perform this operation.

Instead, it is the argument that is invalid (`ServletRequest` is missing an `X-ABAC-Context` header), so `IllegalArgumentException` would be more appropriate.